lib/Models/FunctionParameters/RegionDataParameter.js

```diff
@@ -19,6 +19,7 @@ var RegionTypeParameter = require("./RegionTypeParameter");
  * @constructor
  * @extends FunctionParameter
  *
+ * @param {CatalogFunctionMixin} [catalogFuction] Catalog funciton:
```

Comment:
typo: function
also, why is catalogFunction part of the `options` param?

Reply:
A lot of the js FunctionParameters haven't been ported yet, I did the minimum required to make this work with the new `FunctionParameter` class:

```ts
constructor(
    protected readonly catalogFunction: CatalogFunctionMixin,
    options: Options
  )
```